fix: fixes for the initial version

fhrbata opened the pull request:
This contains three small fixes.

1. We have to distinguish between vulnerable report and esp-idf-sbom error. The esp-idf-sbom tool returned 1 in both cases. This is already handled on the esp-idf-sbom side with internal MR no. 24. With this we can have three states: vulnerable, not vulnerable and error.

2. SBOM_MATTERMOST_WEBHOOK should be tested also for null, not only for unset, because if it's not set in secrets, we would try to sent mattermost message to empty URL.

3. GITHUB_REF_NAME is used by default, but can be overwritten with input "ref" variable. This is needed if one workflow is scanning multiple branches, in which case GITHUB_REF_NAME keeps the same value. GITHUB_REF_NAME is used in user name in the mattermost message, which identifies the repository and branch for which the vulnerability scan was done.

fhrbata committed:
fix: fixes for the initial version

This contains three small fixes.

1. We have to distinguish between vulnerable report and
esp-idf-sbom error. The esp-idf-sbom tool returned 1 in
both cases. This is already handled on the esp-idf-sbom
side. With this we can have three states: vulnerable, not
vulnerable and error.

2. SBOM_MATTERMOST_WEBHOOK should be tested also for null, not only for
unset, because if it's not set in secrets, we would try to sent
mattermost message to empty URL.

3. GITHUB_REF_NAME is used by default, but can be overwritten with
input "ref" variable. This is needed if one workflow is scanning
multiple branches, in which case GITHUB_REF_NAME keeps the same value.
GITHUB_REF_NAME is used in user name in the mattermost message, which
identifies the repository and branch for which the vulnerability scan
was done.

Signed-off-by: Frantisek Hrbata <frantisek.hrbata@espressif.com>

dobairoland reviewed (approved):
LGTM. I've removed the internal link from the description. Please remove it from the commit as well. We don't share those publicly. Thanks!

fhrbata commented:
> LGTM. I've removed the internal link from the description. Please remove it from the commit as well. We don't share those publicly. Thanks!

@dobairoland , thank you for for adjusting the description. I removed it from the commit message. I hope I haven't overlooked anything else. Hopefully this can be merged now. Thank you
